flow360/component/results/case_results.py

```diff
@@ -462,26 +533,104 @@ class SurfaceForcesResultCSVModel(ResultCSVModel):
-    """ForceDistributionResultCSVModel"""
+class PerEntityResultCSVModel(ResultCSVModel):
+    _variables: List[str] = []
+    _x_columns: List[str] = []
+    _entities: List[str] = None
 
-    remote_file_name: str = pd.Field(CaseDownloadable.FORCE_DISTRIBUTION.value, frozen=True)
+    @property
+    def values(self):
+        """
+        Get the current data.
 
-    def wait(self, timeout_minutes=60):
-        """Wait until the Case finishes processing, refresh periodically"""
+        Returns
+        -------
+        dict
+            Dictionary containing the current data.
+        """
+        if self._values is None:
+            super().values
+            self._sum()
+        return super().values
 
-        start_time = time.time()
-        while time.time() - start_time < timeout_minutes * 60:
-            try:
-                self.load_from_remote(log_error=False)
-                return None
-            except CloudFileNotFoundError:
-                pass
-            time.sleep(2)
+    @property
+    def entities(self):
+        """
+        Returns list of entities (boundary names) available for this result
+        """
+        if self._entities is None:
+            pattern = re.compile(rf"(.*)_({'|'.join(self._variables)})$")
+            prefixes = {
+                match.group(1) for col in self.as_dict().keys() if (match := pattern.match(col))
+            }
+            self._entities = sorted(prefixes)
+        return self._entities
+
+    def filter(self, include: list = None, exclude: list = None):
+        """
+        Filters entities based on include and exclude lists.
 
-        raise TimeoutError(
-            "Timeout: post-processing did not finish within the specified timeout period."
+        Parameters
+        ----------
+        include : list or single item, optional
+            List of patterns or single pattern to include.
+        exclude : list or single item, optional
+            List of patterns or single pattern to exclude.
+        """
+        include = (
+            [include] if include is not None and not isinstance(include, list) else include or []
         )
+        exclude = (
+            [exclude] if exclude is not None and not isinstance(exclude, list) else exclude or []
+        )
+
+        include_resolved = list(
+            chain.from_iterable(_find_by_pattern(self.entities, inc) for inc in include)
+        )
+        exclude_resolved = list(
+            chain.from_iterable(_find_by_pattern(self.entities, exc) for exc in exclude)
+        )
+
+        headers = _filter_headers(self._raw_values.keys(), include_resolved, exclude_resolved)
+        self._values = {
+            key: val for key, val in self.as_dict().items() if key in [*headers, *self._x_columns]
+        }
+        self._sum()
+
+    def _sum(self):
+        for variable in self._variables:
+            new_col_name = "total" + variable
+            self._values.pop(new_col_name, None)
```

Comment:
Is there a way to achieve this with regexpr instead of popping it out?

Reply:
addressed in: https://github.com/flexcompute/Flow360/pull/561/commits/7e8543c64dc6fbf027741165109bec18ccd51a6c